src/nupic/math/ArrayAlgo.hpp

```diff
@@ -94,19 +99,25 @@ namespace nupic {
   #elif defined(NTA_ARCH_64)
     #if defined(NTA_OS_LINUX) || defined(NTA_OS_DARWIN)
 
-    __asm__ __volatile__ (
-                         "pushq  %%rbx\n\t"
-
-                         "movl   $1, %%eax\n\t"
-                         "cpuid\n\t"
-                         "movl   %%ecx, %0\n\t"
-                         "movl   %%edx, %1\n\t"
-
-                         "popq  %%rbx\n\t"
-                         : "=c" (c), "=d" (d)
-                         :
-                         :
-			 );
+      __asm__ __volatile__ (
+                            "pushq  %%rbx\n\t"
+
+                            "movl   $1, %%eax\n\t"
+                            "cpuid\n\t"
+                            "movl   %%ecx, %0\n\t"
+                            "movl   %%edx, %1\n\t"
+
+                            "popq  %%rbx\n\t"
+                            : "=c" (c), "=d" (d)
+                            :
+                            :
+                            );
+   #elif defined(NTA_OS_WINDOWS) && defined(NTA_COMPILER_MSVC)
+
+      std::array<int, 4> cpui;
```

Comment:
Are you sure this change retains the same behavior on Windows as the inline assembler?

I'll :+1: this change since this code is already special cased for the platform.